### PR TITLE
Update Binary search tree

### PR DESCRIPTION
**Pull request by weihanglo**
- Fix Issue: left/right tree should reconnect to new node
- Test: unmark .travis.yml test
- Lint files
- Update README.markdown

### CODE REVIEW

File: Binary Search Tree/Solution 1/BinarySearchTree.playground/Sources/BinarySearchTree.swift [hunk 1]
```diff
@@ -106,30 +106,32 @@ extension BinarySearchTree {
...
   @discardableResult public func remove() -> BinarySearchTree? {
     let replacement: BinarySearchTree?
-    
+
     // Replacement for current node can be either biggest one on the left or
     // smallest one on the right, whichever is not nil
-    if let left = left {
-      replacement = left.maximum()
-    } else if let right = right {
+    if let right = right {
```
Comment (kelvinlauKL) [inline Binary Search Tree/Solution 1/BinarySearchTree.playground/Sources/BinarySearchTree.swift]: What's the significance of handling the `right` child before the `left`?

Comment (weihanglo) [inline Binary Search Tree/Solution 1/BinarySearchTree.playground/Sources/BinarySearchTree.swift]: The previous [version](https://github.com/raywenderlich/swift-algorithm-club/blob/ca35d57d14073ac99f6a74c5a2057f1ab7f28c60/Binary%20Search%20Tree/Solution%201/BinarySearchTree.playground/Sources/BinarySearchTree.swift#L135-L139) before 942e3b1 use successor to replace the node with 2 children. The unit tests are also based on this logic. I am kind of lazy to modify all the test cases 😵.

Comment (kelvinlauKL) [inline Binary Search Tree/Solution 1/BinarySearchTree.playground/Sources/BinarySearchTree.swift]: Alright cool. I think this checks out. Merging!

**Comment by kelvinlauKL** [comment]
Thanks @weihanglo!

**Pull request merged**